src/Microsoft.TestPlatform.ObjectModel/TestCase.cs

```diff
@@ -450,6 +329,35 @@ protected override void ProtectedSetPropertyValue(TestProperty property, object
         }
 
         #endregion
+
+        #region ITestCase2 Implementation
+
+        /// <inheritdoc />
+        string ITestCase2.ManagedType
+        {
+            get => GetPropertyValue<string>(TestCaseProperties.ManagedType, null);
+            set => SetPropertyAndResetId(TestCaseProperties.ManagedType, value);
+        }
+
+        /// <inheritdoc />
+        string ITestCase2.ManagedMethod
+        {
+            get => GetPropertyValue<string>(TestCaseProperties.ManagedMethod, null);
+            set => SetPropertyAndResetId(TestCaseProperties.ManagedMethod, value);
+        }
+
+        /// <inheritdoc />
+        string ITestCase2.GetDisplayName()
+        {
+            if (this.HasManagedMethodAndType())
```

Comment:
nit: This method's body can be re-written using the ternary operator, but I acknowledge it's a matter of personal taste.

Reply:
Method removed.